File: Data Structures & Algorithms/validate-parentheses/submission-7.cpp
class Solution {
public:
    bool isValid(string s) {
        stack<char> stack;
        unordered_map<char, char> matches = {{')', '('}, {']', '['}, {'}', '{'}};
        for(char c: s){
            if(matches.count(c)){
                if(!stack.empty() && stack.top()==matches[c]){
                    stack.pop();
                }
                else{
                    return false;
                }
            }
            else{
                stack.push(c);
            }
        }
        if(stack.empty()){
            return true;
        }
        return false;
    }
};

File: Data Structures & Algorithms/counting-bits/submission-7.py
class Solution:
    def countBits(self, n: int) -> List[int]:
        out = []
        count = 0
        for i in range(0,n+1):
            count = 0
            temp = i
            while not i==0:
                count+= i%2
                i = i//2
            out.append(count)
        return out